pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -1085,12 +1084,16 @@ func (p *Planner) enqueueAndSkip(cp *rkev1.RKEControlPlane) error {
 // applied, but output is not yet available for periodic status.
 func (p *Planner) enqueueIfErrWaiting(cp *rkev1.RKEControlPlane, err error) error {
 	if err != nil {
-		w := ErrWaiting("")
-		if errors.As(err, &w) {
-			logrus.Debugf("Enqueuing [%s] because of ErrWaiting: %s", cp.Spec.ClusterName, w.Error())
+		if isErrWaiting(err) {
+			logrus.Tracef("Enqueuing [%s] because of ErrWaiting: %s", cp.Spec.ClusterName, err.Error())
 			return p.enqueueAndSkip(cp)
 		}
 		return err
 	}
 	return nil
 }
+
+func isErrWaiting(err error) bool {
+	var errWaiting ErrWaiting
+	return errors.As(err, &errWaiting)
```

Comment:
```suggestion
	return errors.Is(err, ErrWaiting)
```

Reply:
Unfortunately I can't use `errors.Is` here because it requires also requires a struct, and we aren't uniformly wrapping a single error variable.